### PR TITLE
Annual report scraping and API route

### PR DESCRIPTION
**Pull request by skhan943**
JIRA Ticket
https://skaarkzo-capstone.atlassian.net/browse/SUSTAIN-16

Summary of Changes 
- Annual reports pulled from SEDAR (@zaidokal)
- Combined JSON implementation (@zaidokal)
- Combined scraping service implementation (@zaidokal)
- API route exposed to facilitate scraping data (@skhan943)

### CODE REVIEW

File: app/service/scraper_service.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+def run_subprocess(module_path, *args):
+    command = ["python", "-m", module_path, *args]
+    print(f"Running command: {' '.join(command)}")
+    result = subprocess.run(command, capture_output=True, text=True)
+    print(f"Subprocess completed with return code {result.returncode}")
+    if result.stdout:
+        print("Output:", result.stdout)
+    if result.stderr:
+        print("Error:", result.stderr)
+    return result
```
Comment (rakhshankhan) [inline app/service/scraper_service.py:23]: Can we add some comments here explaining what this is lol

File: app/utilities/sedar_keywords.py
Comment (rakhshankhan) [inline app/utilities/sedar_keywords.py:1]: Can we rename this folder from utilities to "util" or sumn to match with the rest of the naming scheme we got going on.

(i laughed while making this comment)

File: app/scraper/pdf_scraper.py
Comment (rakhshankhan) [inline app/scraper/pdf_scraper.py:1]: Above every scraper (below the imports) can we have a quick summary on what the scraper does and how it's doing it. Also can we comment some of the more complex code in depth more.

File: app/dto/company_scraped_data_dto.py
Comment (rakhshankhan) [inline app/dto/company_scraped_data_dto.py:1]: Don't think we need this DTO anymore

File: app/automation/sedar_automation.py
Comment (rakhshankhan) [inline app/automation/sedar_automation.py:1]: Likewise with the scrapers, can we also give a quick summary on the automation below the imports as well (i.e. how it's doing it, what it's doing and automating, etc.).

File: app/automation/sedar_automation.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+from selenium_stealth import stealth
+
+from app.scraper.pdf_scraper import PDFScraper
+from app.utilities.sedar_keywords import sustainability_keywords
+from app.utilities.sedar_xpaths import *
+
+
+class SedarAutomation:
+    def __init__(self, extract_pdfs: bool = True, pdf_directory: str = 'downloaded_pdfs', temp_directory: str = 'temp_downloads', json_directory: str = 'json_files'):
+        self.base_url = 'https://www.sedarplus.ca'
```
Comment (rakhshankhan) [inline app/automation/sedar_automation.py:19]: The other day we noticed that the Sedar website can go down for maintenence. In the case that happens, how are we error handling when some like that happens?

File: app/scraper/company_website_scraper.py
Comment (rakhshankhan) [inline app/scraper/company_website_scraper.py:1]: Likewise with the PDFScraper, some more commenting on complex code/overall summarization is needed in this.

**Comment by rakhshankhan** [comment]
I wasn't able to comment on the code, but in company_website_scraper, for the keywords, can we also have that in a util file that can be called rather than putting them directly in the scraper? Not sure if that's part of this branch or Sufyan's.

File: app/service/scraper_service.py (hunk 1 above)
Comment (zaidokal) [reply app/service/scraper_service.py]: Added comments to this file

File: app/utilities/sedar_keywords.py
Comment (zaidokal) [reply app/utilities/sedar_keywords.py:1]: Changed utilities folder to util

File: app/automation/sedar_automation.py
Comment (zaidokal) [reply app/automation/sedar_automation.py:1]: Added a description to the sedar automation

File: app/automation/sedar_automation.py (hunk 2 above)
Comment (zaidokal) [reply app/automation/sedar_automation.py:20]: Will have to wait till the website is down to test this out. When it is I can add the functionalities.

File: app/api/scraper/controller/scraper_controller.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+def test_route():
+    return "works!"
+
+SERVICE_SCRIPT = "app.service.scraper_service"
+
+# Define the request body model
+class ProcessRequest(BaseModel):
+    company_name: str
+    website: bool
+    sedar: bool
```
[the remark this answers is not in the document]
Comment (skhan943) [reply app/api/scraper/controller/scraper_controller.py:20]: Done

File: app/api/scraper/controller/scraper_controller.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+from fastapi.responses import JSONResponse
+import subprocess
+import os
+import json
+
+router = APIRouter()
+
+@router.get("/")
+def test_route():
+    return "works!"
```
[the remark this answers is not in the document]
Comment (skhan943) [reply app/api/scraper/controller/scraper_controller.py:12]: Removed

File: app/dto/company_scraped_data_dto.py
Comment (skhan943) [reply app/dto/company_scraped_data_dto.py:1]: Removed

File: app/scraper/company_website_scraper.py
Comment (skhan943) [reply app/scraper/company_website_scraper.py:1]: Done

File: app/scraper/pdf_scraper.py
Comment (skhan943) [reply app/scraper/pdf_scraper.py:1]: Done

**Review by arhamansarii — APPROVED** [review]
Haven't tested on my local machine but looks solid!

**Review by rakhshankhan — APPROVED** [review]

**Pull request merged**